.github/workflows/ci.yml

```diff
@@ -32,6 +32,28 @@ jobs:
           paths_ignore: '["**.md", "**.png", "**.jpg"]'
           do_not_skip: '["workflow_dispatch", "schedule", "push"]'
 
+  report-breaking-changes:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Get modified CRDs
+        id: modified-crds
+        uses: tj-actions/changed-files@v34
+        with:
+          files: |
+            package/crds/**
```

Comment:
So, I believe we assume `make generate` was run and changes committed to the PR. 
I think it is ok to make this assumption since it would be caught by in another job which run a `make check-diff`

Reply:
Yes, correct. As you mentioned, we may not be running the tool on a version reflecting the latest configuration. Currently, it acts more like a non-blocking linter. But when we use the underlying machinery of this flow (collecting the CRDs that have changes to them + running `crddiff` on them) for triggering uptests on the corresponding resources, we will need to revisit the dependencies and the job will start to have blocking semantics (via the tests triggered).